tests/e2e/support/api/davSpaces/spaces.ts

```diff
@@ -13,27 +43,38 @@ export const createFolder = async ({
 }): Promise<void> => {
   const paths = folder.split('/')
 
-  let subFolder = ''
+  let parentFolder = ''
   for (const resource of paths) {
-    const oc10Path = join('remote.php', 'dav', 'files', user.id, subFolder, resource)
-
-    const response = await request({
-      method: 'MKCOL',
-      path: config.ocis
-        ? join(
-            'remote.php',
-            'dav',
-            'spaces',
-            await getPersonalSpaceId({ user }),
-            subFolder,
-            resource
-          )
-        : oc10Path,
-      user: user,
-      formatJson: false
+    const oc10Path = join('remote.php', 'dav', 'files', user.id, parentFolder, resource)
+    // check if the folder exists already or not
+    const folderExists = await doesFolderExists({
+      user,
+      parentFolder,
+      subFolder: resource,
+      oc10Path
     })
+    if (folderExists === false) {
+      const response = await request({
+        method: 'MKCOL',
+        path: config.ocis
+          ? join(
+              'remote.php',
+              'dav',
+              'spaces',
+              await getPersonalSpaceId({ user }),
+              parentFolder,
+              resource
+            )
+          : oc10Path,
+        user: user,
+        formatJson: false
+      })
+      checkResponseStatus(response, 'Failed while creating folder')
+    } else {
+      parentFolder = join(parentFolder, resource)
+      continue
+    }
```

Comment:
I think this can be omitted
```suggestion
    }
```

Reply:
thanks for reporting !!